### PR TITLE
website/integrations: add The Lounge

### PR DESCRIPTION
**Pull request by OddMagnet**
## Details

I just finished setting up LDAP for The Lounge and since there was no Integration Guide yet I figured I'd create one.
I'm very new to Authentik and LDAP, so this guide might need some help/refinement.

---

## Checklist

-   [ ] Local tests pass (`ak test authentik/`)
-   [ ] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [x] The code has been formatted (`make web`)

If applicable

-   [ ] The documentation has been updated
-   [x] The documentation has been formatted (`make website`)

---

## Problems:
I tried running `make website-watch`, but I get this error:
```
[ERROR] [Error: ENOENT: no such file or directory, lstat '/Users/michael/Downloads/authentik/website/static/schema.yaml'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'lstat',
  path: '/Users/michael/Downloads/authentik/website/static/schema.yaml'
}
```
`make website` also gives me an error
```
codespell -w -D - -D .github/codespell-dictionary.txt -I .github/codespell-words.txt -S 'web/src/locales/**' -S 'website/docs/developer-docs/api/reference/**' authentik internal cmd web/src website/src website/blog website/docs website/integrations website/src
make: codespell: No such file or directory
make: *** [lint-codespell] Error 1
```

Since this is just a markdown file I suppose these commands might not be necessary to complete on my system. 
The `make web` mentioned in the checklist worked.

### CODE REVIEW

**Comment by OddMagnet** [comment]
> ### ❌ Deploy Preview for _authentik-docs_ failed.
> Name	Link
> 🔨 Latest commit	[796ab58](https://github.com/goauthentik/authentik/commit/796ab587dbde8b7d528c7ad78f2f2dc9b4bac7fc)
> 🔍 Latest deploy log	https://app.netlify.com/sites/authentik-docs/deploys/672e59c3547ad20008c5fc87

I believe this is not caused by my PR, since the broken links appear to be from somewhere else.

**Comment by tanberry** [comment]
Hi @OddMagnet thanks for this contribution, and apologies for my slow reply! For the formatting and linting, have you tried running `npm run prettier`? I'll start editing the guide, but please let me now about running the linters before pushing.

**Comment by OddMagnet** [comment]
Hey @tanberry I tried running `npm run prettier`, but that fails due to missing the `prettier` script
<details><summary>results</summary>
❯ npm run prettier
npm error Missing script: "prettier"
npm error
npm error To see a list of scripts, run:
npm error   npm run
npm error A complete log of this run can be found in: /Users/michael/.npm/_logs/2024-11-20T13_08_18_604Z-debug-0.log
</details> 

I went back to look at the problems and looked up `codespell`, after installing it via `pip install` I was able to use `make website-install`, `make website-watch` and `make website`

I also noticed that the formatting for the "Note" section was off by a bit and I forgot to add the integration to `sidebarintegrations.js`, oops 😅 

Let me know when I can push the changed

**Comment by tanberry** [comment]
> Hey @tanberry I tried running `npm run prettier`, but that fails due to missing the `prettier` script
> 
> results
> I went back to look at the problems and looked up `codespell`, after installing it via `pip install` I was able to use `make website-install`, `make website-watch` and `make website`
> 
> I also noticed that the formatting for the "Note" section was off by a bit and I forgot to add the integration to `sidebarintegrations.js`, oops 😅
> 
> Let me know when I can push the changed

Great news, thanks for doing that. Sure, after you have run `make website` and you get no (or minimal) errors, and you run `make website-watch` to deploy and checkout your work locally, please do push again to the PR.

**Comment by OddMagnet** [comment]
Done

**Comment by OddMagnet** [comment]
Had the capitalisation of the folder name wrong initially, seems that change didn't make it into my commits, even though everything ran fine locally.

Should be fixed now

File: website/integrations/services/thelounge/index.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+-   `authentik.company` is the FQDN of the authentik install.
+-   `dc=company,dc=com` the Base DN of the LDAP outpost. If you followed the LDAP provider guide this is: `dc=goauthentik,dc=io`
+-   `ldap_bind_user` the username of the desired LDAP Bind User. If you followed the LDAP provider guide this is: `ldapservice`
+
+## LDAP Configuration
+
+### Authentik Configuration
+
+No additional authentik configuration needs to be configured. Follow the LDAP outpost instructions to create an LDAP outpost and configure access via the outpost
```
Comment (tanberry) [inline website/integrations/services/thelounge/index.md:33]:
```suggestion
Follow the [instructions](https://docs.goauthentik.io/docs/add-secure-apps/outposts/#create-and-configure-an-outpost) to create an LDAP outpost and configure access via the outpost. No additional authentik configuration needs to be configured. 
```

File: website/integrations/services/thelounge/index.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+The following placeholders will be used:
+
+-   `authentik.company` is the FQDN of the authentik install.
+-   `dc=company,dc=com` the Base DN of the LDAP outpost. If you followed the LDAP provider guide this is: `dc=goauthentik,dc=io`
+-   `ldap_bind_user` the username of the desired LDAP Bind User. If you followed the LDAP provider guide this is: `ldapservice`
+
+## LDAP Configuration
+
+### Authentik Configuration
```
Comment (tanberry) [inline website/integrations/services/thelounge/index.md:31]:
```suggestion
### authentik Configuration
```

File: website/integrations/services/thelounge/index.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+1. Set `enable` to `true`
+2. Set `url` to `ldap://authentik.company`
+3. Set `primaryKey` to `cn`
+4. In the `searchDN` section make the following changes:
+    1. Set `rootDN` to `cn=ldap_bind_user,ou=users,dc=company,dc=com`
+    2. Set `rootPassword` to the password you have given to the `ldap_bind_user`
+    3. Set `filter` to `(&(objectClass=user)`
+        1. Alternatively, if you want to restrict access by group, you can set it to: `(&(objectClass=user)(memberOf=cn=group_name,ou=groups,dc=ldap,dc=company,dc=com))`
+    4. Set `base` to `dc=ldap,dc=company,dc=com`
+5. Finally, save the `config.js` file and restart The Lounge. You should be able to log in via LDAP now, as long as a user with the same
```
Comment (tanberry) [inline website/integrations/services/thelounge/index.md:48]: @OddMagnet it looks like the rest of the sentence was left off or got lost?

**Review by tanberry — CHANGES_REQUESTED** [review]
A few edits, and the last sentence needs completing, then let's get it merged!

File: website/integrations/services/thelounge/index.md (hunk 3 above)
Comment (OddMagnet) [reply website/integrations/services/thelounge/index.md]: You're right, not sure how that happened 😅 
Fixed it.

**Comment by OddMagnet** [comment]
@tanberry anything else missing?

**Review by tanberry — APPROVED** [review]

**Comment by tanberry** [comment]
> @tanberry anything else missing?

Hi there, the changes look good, thanks! However, the build is still failing on the " lint (prettier-check)" check.

We recently bumped our version on one of the build tools; can you please try running `make website-install` to update the local build tools, and then run `make website` again to run all of the linters and checks?

**Comment by OddMagnet** [comment]
Sure thing @tanberry, I've just pushed the linter fixes.

**Pull request merged**